src/lavinmq/queue/stream_queue.cr

```diff
@@ -0,0 +1,102 @@
+require "./durable_queue"
+require "./stream_queue_message_store"
+require "../client/channel/consumer"
+
+module LavinMQ
+  class StreamQueue < Queue
+    @durable = true
+    @exclusive_consumer = false
+    @no_ack = true
+    getter new_messages : Channel(Bool)
+
+    def initialize(@vhost : VHost, @name : String,
+                   @exclusive = false, @auto_delete = false,
+                   @arguments = Hash(String, AMQP::Field).new)
+      super
```

Comment:
should raise an `ArgumentError` if exclusive or auto_delete is true

Reply:
queue_factory will throw PreconditionFailed if exclusive or auto_delete is true for a stream, is that enough?